src/cmds.rs

```diff
@@ -61,16 +62,6 @@ pub trait CommandResponse {
     }
 }
 
-impl<T> CommandResponse for T
-where
-    T: Read + Write + Sized,
-{
-    fn command_response(&mut self, cmd: &Buffer, res: &mut Buffer) -> io::Result<()> {
-        self.write_all(cmd)?;
-        self.read_exact(res)
-    }
-}
-
```

Comment:
I'm I breaking something by removing this and doing the implementation all in the `lib.rs` `command_response()` function?